setup commit message linting

DaBultz opened the pull request:
This closes #4 

Commitlint did not allow overriding the error messages, however it allowed custom rules (with custom errors) which is the approach I chose. I only implemented 4 rules for now.

it's recommended that commitlint runs on the commit-msg git hook, to make sure this happened husky was added to make sure that the hooks are properly configured and automatically after `npm install`. 

## Rules:

### `docs/no-type` & `docs/no-scope`

These 2 rules disallows the use of the type and scope which is used in the conventional commit standard. these will create an error if they are found in the commit message 

### `docs/header-length` 

This rule will warn if the header is longer than 50 characters long

### `docs/body-blank-line`

This rule will warn if there is no blank line between the body and header

## Questions

1) Are we missing any rules?
2) How much info should each warning/error provide? (multiline, colors, bold, italic and more is possible)
3) Should we extend the contributing guide to include a section about git commit messages?

DaBultz committed:
add husky and commitlint as deps

DaBultz committed:
setup husky

DaBultz committed:
implement commitlint rules

The following rules were implemented:
- docs/no-type: disallows the use of type in the commit message
- docs/no-scope: disallows the use of scope in the commit message
- docs/header-length: max length we recommend for the commit message
  header, this is set to 50 characters long
- docs/body-blank-line: a blank line is recommended between the header
  and the body of the commit message

DaBultz committed:
fix typo in docs/no-scope rule

Co-authored-by: Corey Alexander <coreyja@gmail.com>

jlafayette reviewed (commented):
Thanks for getting this setup!  Is there a step to install the commithook that I'm missing?  I'm seeing an error when I try an invalid message:

```
PS G:\code\docs-new> git commit -v -m "fix: edits"
G:\code\docs-new\node_modules\@commitlint\cli\lib\cli.js:114
        throw err;
        ^

[Error: ENOENT: no such file or directory, open 'G:\code\docs-new\${1}'] {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path: 'G:\\code\\docs-new\\${1}'
}

Node.js v18.4.0
husky - commit-msg hook exited with code 1 (error)
```

If there is, then that might a good thing to add to the README or CONTRIBUTING docs.

All I did was checkout the branch and run `npm install`

DaBultz committed:
fix commitlint on windows

DaBultz committed:
Merge branch 'setup-commitlint' of github.com:DaBultz/docs-new into setup-commitlint

DaBultz committed:
fix commitlint error on windows

coreyja committed:
Add Initial CODEOWNERS File (#8)

* Add Initial CODEOWNERS File

Add a CODEOWNERS file to help facilitate review of PRs

Currently it has the `community-moderators` Team as the default
CODEOWNERS for the entire repo. This will likely change as the Docs site
expands

* Add DaBultz as a Code Owner

jlafayette reviewed (approved):
Thanks for fixing that!  It seems to be working on Windows now

DaBultz committed:
fix body-blank-line lint rule

coreyja reviewed (approved):
💯

DaBultz committed:
Merge branch 'BattlesnakeOfficial:main' into setup-commitlint
